coin.m4

```diff
@@ -1870,7 +1870,7 @@ dnl Make sure the necessary variables exist for each client package.
     ])
 
 dnl Set up command line arguments with DEF_PRIM_ARGS.
-  AC_COIN_DEF_PRIM_ARGS([lapack],yes,yes,no,no)
+  AC_COIN_DEF_PRIM_ARGS([lapack],yes,yes,yes,no)
```

Comment:
We don't do anything with `--with-lapack-cflags`.

Reply:
True, we don't need them for the link check in `CHK_LAPACK` but the user might want to point to a nonstandard place for headers to compile code.

Reply:
But how does the code then know which Lapack implementation it is, i.e., which header file it could include. We don't have a detection or distinction for MKL, OpenBlas, Atlas, etc. so far.

Reply:
The only place this matters is when the user specifies `--with-lapack-cflags` on the command line. We should respect that. This applies in the case (my test MKL installation, for example) where both libraries and header files are in a nonstandard (non-system) location. All the other tests assume we're looking for something that's properly installed in the usual system locations, hence cflags doesn't really come into play.

Reply:
But then your test assumes that Lapack is MKL and that you can include mkl.h. If Lapack is something else, even if that something comes with some headers and corresponding cflags are given, that test won't compile.

I had turned off the `--with-lapack-cflags` in 1837ba4a8 to not give the impression in `configure --help` that the lapack cflags would be used for anything.

Reply:
No argument with what you're saying --- as soon as the user starts using a specific header in their code, they're tied to that implementation of LAPACK. The flip side is that if they do want to choose a specific implementation, specifying `--with-lapack-lflags` and `--with-lapack-cflags` is their only chance. Still, I'm coming around to your point of view. `CHK_LAPACK` does the right thing (link check only) for COIN code, which is careful not to require a specific implementation. If the user is writing their own code and wants a specific implementation, it should be a separate project and they can configure as they like, using `CHK_LIBHDR` & friends. I'll disable `--with-lapack-cflags`.